premailer/tests/test_cache.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import absolute_import, unicode_literals
+
+import os
+import unittest
+import imp
+
+import cachetools
+
+
+class TestFunctionCache(unittest.TestCase):
+    def tearDown(self):
+        for key in (
+            "PREMAILER_CACHE",
+            "PREMAILER_CACHE_MAXSIZE",
+            "PREMAILER_CACHE_TTL",
+        ):
+            try:
+                del os.environ[key]
+            except KeyError:
+                pass
+
+    def test_unknown_cache(self):
+        os.environ["PREMAILER_CACHE"] = "UNKNOWN"
+
+        with self.assertRaises(Exception) as assert_context:
+            cache_fd = open("premailer/cache.py", "r")
+            imp.load_source("cache.py", "premailer/cache.py", cache_fd)
```

Comment:
Will this leave an open file descriptor here? Ever? Why not:
```python
with open("premailer/cache.py") as fd:
    imp.load_source("cache.py", "premailer/cache.py", fd)
```

Reply:
Nice catch. Thanks for pointing it out.
I actually don't even need it. Removed.